app/models/piece.rb

```diff
@@ -36,7 +44,11 @@ def captured?
   def valid_move?(to_row, to_col)
     return false if move_nil?(to_row, to_col)
     return false if move_out_of_bounds?(to_row, to_col)
-    return false if move_destination_same_side?(to_row, to_col)
+    return false if move_destination_ally?(to_row, to_col)
+
+    # until have Knight::move_legal?
+    return true if type == "Knight"
```

Comment:
Instead of having this line, you could add some filler code directly into `knight.rb`. Something like
```
class Knight < Piece
  def move_legal?(...)
    return true # This is just filler code, replace with actual code
...
```